### PR TITLE
Upgrade Redux, config for XO

### PR DESCRIPTION
**Pull request by wmertens**
Just a few changes to make it work for me.

* Store creation didn't work
* NPM v3 was complaining about peers.
* XO wasn't configured for xo-linter in my editor

### CODE REVIEW

File: src/client/makeStore.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  if (!__PRODUCTION__) {
+    const {ensureState} = require('redux-optimistic-ui');
+
+    reduxRouterMiddleware.listenForReplays(store, state => ensureState(state).get('routing'));
+  }
+
+  return store;
+};
+
+export default makeStore;
```
Comment (Bartekus) [inline src/client/makeStore.js]: Looks good, I'm gonna test this first thing in the morning... right now I'm passing out :sleeping:

File: package.json [hunk 2]
```diff
@@ -107,5 +109,18 @@
...
+      "browser",
+      "node",
+      "webpack"
+    ],
+    "extends": "xo-react",
+    "globals": [
+      "__PRODUCTION__"
+    ]
   }
 }
```
Comment (Bartekus) [inline package.json]: I'm not a big fan of xo because of this specific aspect of it... polluting the package.json
What are your thoughts on eslint?

Comment (wmertens) [inline package.json]: XO is a wrapper around eslint, making it easier to configure and providing linter interfaces that can call `--fix` from the editor. Love atom-linter-xo…

I don't see this as polluting package.json, it accurately describes the project coding style in a file that is always there. 

Anyway, if you don't like that there's the manual way as described at https://github.com/sindresorhus/xo#why-not-eslint

**Comment by Bartekus** [comment]
Well you convinced me, after all my position was a bit misplaced and between you and Matt, this seems very reasonable after all. Alright... I'll be back in few hours, but so far I like what I see! Cheers for now!

**Comment by mattkrick** [comment]
LGTM, but my internet is still too poor to do a git pull & test it out :sob:. 
@Bartekus I feel ya on the package json, but i'm not sure a dotfile is any cleaner, although I guess we could move that way if we wanted. XO is 98% eslint, with a nice little fix feature. 
PS, if this looks good to you, i'll blindly merge, since we can't do any worse than the currently broken HEAD :laughing:

**Comment by wmertens** [comment]
Just noticed I missed the __CLIENT__ global, added and force-pushed it.

**Comment by wmertens** [comment]
Also fixed most of the lint errors. Any chance of tabs instead of spaces, and no semicolons? ;)

**Comment by mattkrick** [comment]
@wmertens there was some blog awhile back that stated my sentiments exactly. I don't care what we use, as long as we all use the same thing. The vast majority of stuff I see uses spaces & semicolons so I've been sticking with that. Same thing about `import { foo }` vs `import {foo}`. The only people who include spaces work at facebook...

**Comment by wmertens** [comment]
Okay, for me the big one is tabs (http://lea.verou.me/2012/01/why-tabs-are-clearly-superior/), so hopefully you'll see that reaching critical mass at some point ;)

**Comment by mattkrick** [comment]
just about every editor or IDE will convert tabs to X spaces, so i dunno if it'll gain any momentum. eslint defaults to 2 spaces, doesnt it? 

PS, if someone else can confirm that works I'll do a merge & get this unborked :)

**Comment by kurtaschliman** [comment]
I pulled these changes in and they worked for me. I did not do a full fresh install, but a good ole `rm -rf node_modules && npm install` and the errors were no more.

**Comment by mattkrick** [comment]
that's good nuff for me, thanks @wmertens!

**Pull request merged**